builtin/logical/database/path_config_connection.go

```diff
@@ -210,44 +205,54 @@ func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 		}
 
 		verifyConnection := data.Get("verify_connection").(bool)
-
 		allowedRoles := data.Get("allowed_roles").([]string)
+		rootRotationStatements := data.Get("root_rotation_statements").([]string)
 
 		// Remove these entries from the data before we store it keyed under
 		// ConnectionDetails.
 		delete(data.Raw, "name")
 		delete(data.Raw, "plugin_name")
 		delete(data.Raw, "allowed_roles")
 		delete(data.Raw, "verify_connection")
+		delete(data.Raw, "root_rotation_statements")
 
-		config := &DatabaseConfig{
-			ConnectionDetails: data.Raw,
-			PluginName:        pluginName,
-			AllowedRoles:      allowedRoles,
-		}
-
-		db, err := dbplugin.PluginFactory(ctx, config.PluginName, b.System(), b.logger)
+		// Create a database plugin and initialize it. This instance is not
+		// going to be used and is initialized just to ensure all parameters
+		// are valid and the connection is verified, if requested.
+		db, err := dbplugin.PluginFactory(ctx, pluginName, b.System(), b.logger)
```

Comment:
If verifying the credentials, should we also attempt to verify the root rotation logic by rotating right here? That way if something is wrong they discover it right at config time. (We could gate this behavior behind another bool, too.)

Reply:
I think the easiest thing to do here is leave this option to the outside caller.  They can always config the connection, verify, and rotate.